### ISSUE TITLE
Random Angular Velocity in Particles2D is not random

### ISSUE THREAD

**Issue by Arnklit** [godotengine/godot#55585]
### Godot version

3.4.Stable

### System information

Windows 10, GLES3, GTX 980Ti

### Issue description

Random Angular Velocity in Particles2D is not random. Particles rotate based on their Y position in the shape extends. Note how particles in the top rotate clockwise and in the bottom counter clockwise.

![AWiscUW7kP](https://user-images.githubusercontent.com/4955051/144599104-f6d93c3d-4138-41d2-83f0-b303a56830d2.gif)

### Steps to reproduce

1. Create a particles2D. set the texture to the icon.
2. Set a Particles process material.
3. Set the Emission Shape to Box
4. set the Angular Velocity to 100 and the Angular Velocity Random to 1.0.

### Minimal reproduction project

[particles2d_angular_random_bug.zip](https://github.com/godotengine/godot/files/7649409/particles2d_angular_random_bug.zip)

**Comment by pycbouh** [comment]
I've tested on 3.4.stable, and can confirm. Interesting enough, CPUParticles2D have similar math, but it works correctly

```cpp
base_angle += p.custom[1] * lifetime * (parameters[PARAM_ANGULAR_VELOCITY] + tex_angular_velocity) * Math::lerp(1.0f, rand_from_seed(alt_seed) * 2.0f - 1.0f, randomness[PARAM_ANGULAR_VELOCITY]);
```

vs

```glsl
base_angle += CUSTOM.y * LIFETIME * (angular_velocity + tex_angular_velocity) * mix(1.0, rand_from_seed(alt_seed) * 2.0 - 1.0, angular_velocity_random);
```

**Comment by akien-mga** [comment]
CC @RPicster who's been working on these recently.

**Comment by RPicster** [comment]
I observed this behaviour recently on other parameters and might have an idea where to start looking. Thanks for tagging me.

**Comment by Calinou** [comment]
Is this issue reproducible on the `master` branch?

**Comment by RPicster** [comment]
No, the shader structure is different in master.

**Comment by akien-mga** [comment]
Fixed by #55607.
